app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkItemMenu.kt

```diff
@@ -25,59 +25,77 @@ class BookmarkItemMenu(
-            if (itemType != BookmarkNodeType.SEPARATOR) {
-                TextMenuCandidate(
-                    text = context.getString(R.string.bookmark_menu_edit_button)
-                ) {
-                    onItemTapped.invoke(Item.Edit)
-                }
-            } else {
-                null
-            },
-            if (itemType == BookmarkNodeType.ITEM) {
-                TextMenuCandidate(
-                    text = context.getString(R.string.bookmark_menu_copy_button)
-                ) {
-                    onItemTapped.invoke(Item.Copy)
-                }
-            } else {
-                null
-            },
-            if (itemType == BookmarkNodeType.ITEM) {
-                TextMenuCandidate(
-                    text = context.getString(R.string.bookmark_menu_share_button)
-                ) {
-                    onItemTapped.invoke(Item.Share)
-                }
-            } else {
-                null
-            },
-            if (itemType == BookmarkNodeType.ITEM) {
-                TextMenuCandidate(
-                    text = context.getString(R.string.bookmark_menu_open_in_new_tab_button)
-                ) {
-                    onItemTapped.invoke(Item.OpenInNewTab)
-                }
-            } else {
-                null
-            },
-            if (itemType == BookmarkNodeType.ITEM) {
-                TextMenuCandidate(
-                    text = context.getString(R.string.bookmark_menu_open_in_private_tab_button)
-                ) {
-                    onItemTapped.invoke(Item.OpenInPrivateTab)
-                }
-            } else {
-                null
-            },
+            processItem(
+                itemType,
+                listOf(BookmarkNodeType.ITEM, BookmarkNodeType.FOLDER),
+                context.getString(R.string.bookmark_menu_edit_button),
+                Item.Edit
+            ),
+            processItem(
+                itemType,
+                BookmarkNodeType.ITEM,
+                context.getString(R.string.bookmark_menu_copy_button),
+                Item.Copy
+            ),
+            processItem(
+                itemType,
+                BookmarkNodeType.ITEM,
+                context.getString(R.string.bookmark_menu_share_button),
+                Item.Share
+            ),
+            processItem(
+                itemType,
+                BookmarkNodeType.ITEM,
+                context.getString(R.string.bookmark_menu_open_in_new_tab_button),
+                Item.OpenInNewTab
+            ),
+            processItem(
+                itemType,
+                BookmarkNodeType.ITEM,
+                context.getString(R.string.bookmark_menu_open_in_private_tab_button),
+                Item.OpenInPrivateTab
+            ),
+            processItem(
+                itemType,
+                BookmarkNodeType.FOLDER,
+                context.getString(R.string.bookmark_menu_open_all_in_tabs_button),
```

Comment:
Partially done in eb07f89 (this commit will be squash when approved).

I only check one level of children in the display or not check. I chose this, because we are forced to use a coroutine since `getTree` is async and on slow android with 10 levels and ~20 items was quite bugging because coroutine has not yet finished. Keeping one level seems me not that bad. (I will fully check all level when user has clicked on open all to show a war message if >15 as requested in others reviews.)

So:
```
root
 |-> folder1
 |    | -> item1
 |    |-> folder2
 |         | -> item2
 | -> emptyfolder1
       | -> emptyfolder2
```
folder1, folder2 & emptyfolder1 show option. Not emptyfolder2.

Reply:
That is great! Thank you!